crates/turbo-tasks-memory/src/task.rs

```diff
@@ -387,6 +418,8 @@ use self::meta_state::{
     FullTaskWriteGuard, TaskMetaState, TaskMetaStateReadGuard, TaskMetaStateWriteGuard,
 };
 
+const SCOPE_OPTIMIZATION_THRESHOLD: usize = 255;
```

Comment:
Add a comment to explain what this const controls. Also, perhaps `(2 << 7) - 1` would be clearer here?